### ISSUE TITLE
ssh_config aliases do not work with jsch-ssh

### ISSUE THREAD

**Issue by jquick** [rundeck/rundeck#2107]
# Enhancement
**My Rundeck detail**

* Rundeck version: 2.6.9-1
* install type: deb (rpm,deb,war,launcher)
* OS Name/version: osx/10.12

**Expected Behavior**

Should be allowed to load in ssh_config to either jsch-ssh or local-ssh-agent

**Actual Behavior**

It does not work. Cannot see the config.

**How to reproduce Behavior**

Setup a alias in /etc/ssh/ssh_confg. Works fine with CLI:
~~~
root@rundeck:~# ssh jquick@worker-cluster -i /var/lib/rundeck/.ssh/id_rsa
Welcome to Ubuntu 12.04.5 LTS (GNU/Linux 3.2.0-39-virtual x86_64)
...
~~~
Rundeck cannot see the alias:
~~~
[workflow] beginExecuteNodeStep(worker-cluster): NodeDispatch: com.dtolabs.rundeck.execution.ExecutionItemFactory$4@620f765a
Using ssh keyfile: /var/lib/rundeck/.ssh/id_rsa
Starting SSH Connection: jquick@worker-cluster (worker-cluster)
Using private key file: /var/lib/rundeck/.ssh/id_rsa
Set timeout to 0
Connecting to worker-cluster:22
Connecting to worker-cluster port 22
worker-cluster
Failed: HostNotFound: worker-cluster
~~~

Basically we are using a ssh alias with a command that dynamically specifies a node with the least usage. I did try to use the script-exec and it was resolving properly but was running into other issues (script not ending after execution, etc). I am hoping to get this to work with the stock ssh plugin. 

Is there a way to get our config to be loaded? or somehow to get this to work?

Another workaround is if there is the possibility of a "hostname: \`dynamic-hostname-script\`" or something possible for the node.

Thanks for any advice!

**Comment by jquick** [comment]
I did see the ssh-example-1.0-plugin, but wanted to make sure im not missing something simple before I built something custom.

**Comment by jquick** [comment]
Tried doing something like ssh-config-ProxyConnect and also tried the local-ssh-client but no go.
https://github.com/rundeck/rundeck/issues/1523 looks like this is related. Looks like doing:
{code}
script-exec="/usr/bin/ssh -F /etc/ssh/ssh_config ${node.username}@${node.hostname} -- ${exec.command}"
{code}
May be the best bet. This seems to work except it keeps running after everything has been executed.

**Comment by jquick** [comment]
Looks like the issue with the script not returning was the masking of the hostname. Resolving the proper hostname before connection is now working as expected.

**Comment by jquick** [comment]
Ok got it working via:
```
  <node name="worker-cluster" description="worker-cluster" tags="" hostname="worker-cluster" osFamily="unix" username="jquick"
        node-executor="script-exec"
        script-exec-shell="bash -c"
        script-exec="/usr/bin/ssh -o StrictHostKeyChecking=no ${node.username}@$(ballast-exec ${node.hostname} ${job.execid}) -- 'echo Load Balancer Connected: $(hostname -A); ${exec.command}'"
        file-copier="script-copy"
        script-copy-shell="bash -c"
        script-copy="scp ${file-copy.file} ${node.username}@$(ballast-exec ${node.hostname} ${job.execid}):${file-copy.destination}"
  />
```
